pkg/render/nonclusterhost/nonclusterhost.go

```diff
@@ -137,6 +137,18 @@ func (c *nonClusterHostComponent) clusterRole() *rbacv1.ClusterRole {
 			Resources: []string{"nodes"},
 			Verbs:     []string{"get", "list", "watch"},
 		},
+		{
+			// For non-cluster host to get tigera-ca-bundle config map.
+			APIGroups: []string{""},
+			Resources: []string{"configmaps"},
+			Verbs:     []string{"get"},
+		},
+		{
+			// For non-cluster host to request a operator signed certificate.
+			APIGroups: []string{"certificates.k8s.io"},
+			Resources: []string{"certificatesigningrequests"},
+			Verbs:     []string{"create", "delete", "list", "watch"},
```

Comment:
Nit: I don't think it needs delete. I thought these are automatically removed after some time by k8s.

Reply:
The `delete` verb is removed. It is needed by key-cert-provisioner to delete any existing denied CSRs with the same name. For non-cluster host case, I believe it would be a programming error if a CSR got denied.